lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -411,53 +398,85 @@ def calculate_wet_bulb_temperature(self, temperature, relative_humidity,
                 Cube of air pressures (Pa).
 
         Returns:
-            wbt (iris.cube.Cube):
+            iris.cube.Cube:
                 Cube of wet bulb temperature (K).
 
         """
-        precision = np.full(temperature.data.shape, self.precision)
-
-        # Set units of input diagnostics.
         relative_humidity.convert_units(1)
         pressure.convert_units('Pa')
         temperature.convert_units('K')
 
+        precision = np.full(temperature.data.shape, self.precision)
+        wbt = temperature.copy()
+        wbt.rename('wet_bulb_temperature')
+
+        latent_heat = Utilities.latent_heat_of_condensation(temperature)
+
+        wbt_data = self._calculate_wbt(
+            latent_heat.data, precision, pressure.data,
+            relative_humidity.data, temperature.data)
+
+        wbt.data = wbt_data
```

Comment:
I would recommend getting rid of lines 410-411 and changing the call here to:
```
wbt = temperature.copy(data=wbt_data)
wbt.rename('wet_bulb_temperature')
```
This is just a bit tidier.

Reply:
Also calculating `precision` just before the `_calculate_wbt` call.  Declaring the variable immediately before it's used is nicer to read.

Reply:
changed the cube/copying/naming

also

moved precision inside the function.